rust/segment/src/local_hnsw.rs

```diff
@@ -520,6 +521,8 @@ impl LocalHnswSegmentWriter {
             return Ok(next_label);
         }
         let mut max_seq_id = u64::MIN;
+        // In order to insert into hnsw index in parallel, we need to collect all the embeddings
+        let mut hsnw_batch = Vec::with_capacity(log_chunk.len());
         for (log, _) in log_chunk.iter() {
```

Comment:
I am guessing you can't par_iter() on log_chunk?

Reply:
I need the labels that get generated - so kept this with both